src/sparseml/pytorch/torchvision/train.py

```diff
@@ -886,7 +934,7 @@ def new_func(*args, **kwargs):
     type=str,
     help="json parsable dict of recipe variable names to values to overwrite with",
 )
-@click.option("--dataset-path", required=True, type=str, help="dataset path")
```

Comment:
Maybe it would a good idea to check whether the user has specified either a sufficient set of arguments to create a dataset from deeplake or from generic dataset, and throw error if both pathways or none of them is enabled.